File: app/src/main/java/com/example/chenyong/testdemo/MainActivity.java
package com.example.chenyong.testdemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.cy.customwidget.NumPercent;

public class MainActivity extends AppCompatActivity {

    private int mProgressValue = 1;
    private int mTotalValue = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final NumPercent numPercent = findViewById(R.id.num_percent);
        numPercent.setTotalValue(mTotalValue);
        numPercent.setProgressValue(mProgressValue);

        findViewById(R.id.btn_progress).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                numPercent.setProgressValue(++mProgressValue);
            }
        });

        findViewById(R.id.btn_total).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                numPercent.setTotalValue(++mTotalValue);
            }
        });
    }
}

File: PhoneAndSms/src/main/java/com/kungeek/phoneandsms/MainActivity.kt
package com.kungeek.phoneandsms

import android.content.Context
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.telephony.TelephonyManager
import android.util.Log
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val textView = findViewById<TextView>(R.id.tv_phone)
        findViewById<Button>(R.id.btn_phone).setOnClickListener {
            val telephonyManager = getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
            val phone = telephonyManager.line1Number
            Log.d("phoneNumber", phone)
            textView.text = phone
        }
    }
}
